### PR TITLE
📝 Contributors guide

### LINKED ISSUE #10: Implement Software Development Lifecycle
- [x] Create the standard set of git repo labels for all branch types: feature, refactor, devops, bug, documentation, test
- [x] Create a git repo pull request template that automatically populates PR descriptions when the PR is opened
- [x] Create a README with a contributors guide OR create a separate CONTRIBUTING.md for the repo so that developers understand what the software development lifecycle is
- [x] Add Github release workflow so that IGs can be versioned and archived via releases
- [x] Setup deploy previews for the IG website so that every PR has a preview of the IG with the changes introduced in the PR
- [x] (Maybe) Setup IG validation that runs on every pushed commit

### PR DESCRIPTION
**Pull request by znatty22**
# Motivation
Multiple people are going to start contributing to this repository but we don't have guidelines on the software development lifecycle for people to follow.

# Approach
Create a contributors guide via a markdown doc that new developers can follow to understand the basic Github workflow of authoring commits, reviewing code, and making releases.

Closes #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+This repository is currently under development.
+Feel free to [take a look](https://torstees.github.io/ncpi-fhir-ig-2/) at the working version.
+
+## 👨🏻‍💻 Contributors Guide
+
+If you would like to contribute to this project, please take a look at the 
+software development lifecycle and standard practices for this project
+[here](./.github/contributing.md).
 
-This repository is currently under development. Feel free to [take a look](https://torstees.github.io/ncpi-fhir-ig-2/) at the working version.
```
Comment (znatty22) [inline README.md:4]: @torstees Is this line still relevant or should we remove it?

**Review by aw3334 — DISMISSED** [review]

File: .github/contributing.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+**✨ Add new birthSex field to Patient**
+
+### Description
+When you open a pull request you will notice that the description is populated
+with a "Motivation" and "Approach" section. Please be sure to fill these out 
+so that reviewers can quickly and easily understand the purpose of the PR and 
+the changes.
+
+## Squash and Merge
```
Comment (wnkhan) [inline .github/contributing.md:119]: The consensus in the NCPI code-a-thon is that we avoid squashing to preserve commit history.

File: README.md (hunk 1 above)
Comment (znatty22) [reply README.md:4]: Eric will remove this

File: .github/contributing.md (hunk 2 above)
Comment (znatty22) [reply .github/contributing.md:119]: We decided that preserving nitty gritty commit history is not important and we believe that squash merge will make rebasing a lot easier

File: .github/contributing.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+# to the change
+git checkout -b add-patient-birth-sex 
+```
+
+### Commit the change 
+
+After you've made the change you want on your local machine its time 
+to commit that change to the project repo. 
+
+#### Emoji Commits
```
Comment (wnkhan) [inline .github/contributing.md:57]: Update wording to indicate that the gitmoji's are optional or at least strict adherence won't be required. Will still be included to give more robust search options for browsing commit history.

**Review by torstees — APPROVED** [review]
this looks good to me!

**Pull request merged**